### PR TITLE
Expanded patterns on debug and session, added more info to debug

### PR DESCRIPTION
**Pull request by honza-kasik**
Added `<path>/*` patterns to session and debug to allow testing URL context delimiters in clustered environment

Added extra path info and query string output to debug servlet for the same reason.

### CODE REVIEW

**Comment by rhusar** [comment]
@tommaso-borgato Do you also want to add your review?

File: clusterbench-ee10-web/src/main/java/org/jboss/test/clusterbench/web/session/HttpSessionServlet.java [hunk 1]
```diff
@@ -23,7 +23,7 @@
 /**
  * @author Radoslav Husar
  */
-@WebServlet(name = "HttpSessionServlet", urlPatterns = { "/session" })
+@WebServlet(name = "HttpSessionServlet", urlPatterns = { "/session", "/session/*" })
```
Comment (tommaso-borgato) [inline clusterbench-ee10-web/src/main/java/org/jboss/test/clusterbench/web/session/HttpSessionServlet.java:26]: @honza-kasik what is the reason for this change?

File: clusterbench-ee10-web/src/main/java/org/jboss/test/clusterbench/web/debug/DebugServlet.java [hunk 2]
```diff
@@ -31,7 +31,7 @@
 /**
  * @author Radoslav Husar
  */
-@WebServlet(name = "DebugServlet", urlPatterns = { "/debug" })
+@WebServlet(name = "DebugServlet", urlPatterns = { "/debug", "/debug/*" })
```
Comment (tommaso-borgato) [inline clusterbench-ee10-web/src/main/java/org/jboss/test/clusterbench/web/debug/DebugServlet.java:34]: @honza-kasik what is the reason for this change?

File: clusterbench-common/src/main/java/org/jboss/test/clusterbench/common/debug/AbstractCommonDebugServlet.java [hunk 3]
```diff
@@ -84,6 +87,13 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IO
 
         // Request URI
         out.println("Request URI: " + req.getRequestURI());
+        // Query path of the URL
+        out.println("Query string: " + req.getQueryString());
+        // Query path of the URL decoded
+        out.println("Query string UTF-8 decoded: " + ((req.getQueryString() == null) ? "null" :
+                                                    URLDecoder.decode(req.getQueryString(), StandardCharsets.UTF_8)));
+        // Extra path information
+        out.println("Path info: " + req.getPathInfo());
```
Comment (tommaso-borgato) [inline clusterbench-common/src/main/java/org/jboss/test/clusterbench/common/debug/AbstractCommonDebugServlet.java:96]: @honza-kasik what is the reason for this changes?

Comment (honza-kasik) [reply clusterbench-common/src/main/java/org/jboss/test/clusterbench/common/debug/AbstractCommonDebugServlet.java:96]: This is here to server context delimiter tests in mod_cluster testsuite. To my knowledge it is mainly used to ensure that the context delimiters are properly recognized.

File: clusterbench-ee10-web/src/main/java/org/jboss/test/clusterbench/web/debug/DebugServlet.java (hunk 2 above)
Comment (honza-kasik) [reply clusterbench-ee10-web/src/main/java/org/jboss/test/clusterbench/web/debug/DebugServlet.java:34]: This was added so "sub paths" will access this servlet as well and debug data can be collected even with longer paths.

File: clusterbench-ee10-web/src/main/java/org/jboss/test/clusterbench/web/session/HttpSessionServlet.java (hunk 1 above)
Comment (honza-kasik) [reply clusterbench-ee10-web/src/main/java/org/jboss/test/clusterbench/web/session/HttpSessionServlet.java:26]: Nope, it shouldn't affect any existing behaviour. This only extends path to include all subpaths. This was needed by context delimiter tests again. To my knowledge it is verified that various queries won't break session integrity. I'll try to run some tests.

Comment (honza-kasik) [reply clusterbench-ee10-web/src/main/java/org/jboss/test/clusterbench/web/session/HttpSessionServlet.java:26]: I tried to run a configuration and it went OK - same as with upstream one.

**Comment by honza-kasik** [comment]
@rhusar @tommaso-borgato Thanks guys for the review! Can this be merged now please?

**Comment by tommaso-borgato** [comment]
@honza-kasik since you run clustering test successfully, I guess we can merge.... @rhusar can you proceed and merge this one?

**Comment by rhusar** [comment]
Thanks @honza-kasik @tommaso-borgato ! Merging.
